[SSM-3033] Added a multipart file upload example

denisahearn opened the pull request:
## What / Why?
Code example demonstrating how to upload a file to FieldAgent using a multipart upload.

### Screenshot(s)

### JIRA Link
https://sentera.atlassian.net/browse/SSM-3033

## QA Strategy
- [ ] Merge Latest Master
- [ ] Regression Test?
- [ ] Test New Feature?

denisahearn committed:
Added a multipart file upload example

denisahearn committed:
Removed temporary local FieldAgent host URL

nickb27 commented on README.md:
```diff
@@ -1,16 +1,30 @@
-## Workflow
-There are three basic steps for uploading a file and using it with Sentera FieldAgent:
+## API Credentials
+To run these file upload examples, you must first obtain an access token for your FieldAgent user that will be used to authenticate your requests to the FieldAgent GraphQL API. See https://api.sentera.com/api/getting_started/authentication_and_authorization.html for details on obtaining an API access token. Once you have a valid access token, paste it into a file named `auth_token.txt` that is lo ...
```
```suggestion
To run these file upload examples, you must first obtain an access token for your FieldAgent user that will be used to authenticate your requests to the FieldAgent GraphQL API. See https://api.sentera.com/api/getting_started/authentication_and_authorization.html for details on obtaining an API access token. Once you have a valid access token, paste it into a file named `fieldagent_access_token.txt` that is located in the same directory as the code examples.
```

nickb27 commented on fieldagent_access_token.txt.example:
```diff
@@ -0,0 +1 @@
+<Your FieldAgent access token goes here>
```
Can you add an example token here?

denisahearn replied:
How's this? https://github.com/SenteraLLC/file-upload-examples/pull/2/commits/2efaddf02fe97a919828a444772e7861ba91f626

nickb27 replied:
Lovely!  Thanks!

denisahearn committed:
Update README.md

Co-authored-by: Nick Brill <nick.brill@sentera.com>

denisahearn committed:
Added an example FieldAgent access token to fieldagent_access_token.txt.example

denisahearn committed:
Merge branch 'ssm-3033-multipart-file-upload-example' of github.com:SenteraLLC/file-upload-examples into ssm-3033-multipart-file-upload-example

denisahearn committed:
Add FIELDAGENT_SERVER environment variable

denisahearn commented:
@nickb27 I added a `FieldAgent Server` section to `README.md` to call out the FieldAgent server URL that the upload scripts use. I also added the use of a `FIELDAGENT_SERVER` environment variable that can be specified on the command line to use a different FieldAgent server than the one coded in `utils.rb`. This way someone can specify a different URL without having to modify the source code.

https://github.com/SenteraLLC/file-upload-examples/pull/2/commits/cfe5f9e88e798f30b4026bf28a74dd9a7241d9da

Please have a look and let me know what you think.

denisahearn committed:
Added FIELDAGENT_ACCESS_TOKEN environment variable

denisahearn committed:
Use pry-byebug

denisahearn commented on utils.rb:
This file contains Ruby code shared between `file_upload.rb` and `multipart_file_upload.rb`

jdalt commented on file_upload.rb:
```diff
@@ -1,55 +1,44 @@
 # frozen_string_literal: true
```
```suggestion
#!/usr/bin/env ruby

# frozen_string_literal: true
```
This would allow you to run the script without having to pass the ruby interpreter on the commandline.

jdalt commented on multipart_file_upload.rb:
```diff
@@ -0,0 +1,303 @@
+
+  # Compute the number of parts based on 5MB chunks
+  file_size_bytes = File.size(file_path)
+  part_size_bytes = 5 * 1024 * 1024 # 5 megabytes is the smaller part size AWS S3 permits
```
```suggestion
  part_size_bytes = 5 * 1024 * 1024 # 5 megabytes is the smallest part size AWS S3 permits
```
superlative?

jdalt commented on multipart_file_upload.rb:
```diff
@@ -0,0 +1,303 @@
+# frozen_string_literal: true
```
```suggestion
#!/usr/bin/env ruby

# frozen_string_literal: true
```

jdalt commented on multipart_file_upload.rb:
```diff
@@ -0,0 +1,303 @@
+def complete_multipart_file_upload(parts, s3_key, upload_id)
+  puts 'Complete multipart file upload'
+
+  parts_gql = '[' + parts.reduce("") do |memo, part|
+    memo += %Q|{part_number: #{part[:part_number]}, etag: "#{part[:etag]}"}|
+    memo
+  end + ']'
```
Would it be easier to write a query with a variable so that you didn't have to serialize GQL (could just pass a more normal js array into variable)?

denisahearn replied:
Thanks for the suggestion, that cleaned things up nicely! https://github.com/SenteraLLC/file-upload-examples/pull/2/commits/a7cb16b058a554f14ebd94826d6a1812ecf023e7

denisahearn committed:
Update file_upload.rb

Co-authored-by: Jacob Dalton <jacobrdalton@gmail.com>

denisahearn committed:
Update multipart_file_upload.rb

Co-authored-by: Jacob Dalton <jacobrdalton@gmail.com>

denisahearn committed:
Update multipart_file_upload.rb

Co-authored-by: Jacob Dalton <jacobrdalton@gmail.com>

denisahearn committed:
Use GQL variables
